services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/RequestModifierUtils.java

```diff
@@ -30,32 +26,22 @@
 import software.amazon.awssdk.utils.Validate;
 
 /**
- * This interceptor transforms the {@code sourceBucket}, {@code sourceKey}, and {@code sourceVersionId} parameters for
- * {@link CopyObjectRequest} and {@link UploadPartCopyRequest} into a {@code copySource} parameter. The logic needed to
- * construct a {@code copySource} can be considered non-trivial, so this interceptor facilitates allowing users to
- * use higher-level constructs that more closely match other APIs, like {@link PutObjectRequest}. Additionally, this
- * interceptor is responsible for URL encoding the relevant portions of the {@code copySource} value.
+ * This class transforms specific request parameter into a custom parameter. The logic needed to
+ * construct parameters like {@code copySource} which is non-trivial, so this class facilitates the use of
+ * higher-level constructs that more closely match other APIs, such as {@link PutObjectRequest}.
  * <p>
- * <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html#API_CopyObject_RequestParameters">API_CopyObject_RequestParameters</a>
+ * For more information, see:
+ * <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html#API_CopyObject_RequestParameters">CopyObject Request Parameters</a>
  * <p>
- * <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPartCopy.html#API_UploadPartCopy_RequestParameters">API_UploadPartCopy_RequestParameters</a>
+ * <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPartCopy.html#API_UploadPartCopy_RequestParameters">UploadPartCopy Request Parameters</a>
  */
 @SdkInternalApi
-public final class CopySourceInterceptor implements ExecutionInterceptor {
+public final class RequestModifierUtils  {
```

Comment:
It would be good to keep a reference to `CopySource` in this class name, `CopySourceModifierUtils` maybe?

Reply:
I updated the name as CustomRequestTransformerUtils since in future if we need to do similar customization to any API then we can just add an API in this class.
Thus this class will have broader scope of all the RequestTransformer rather than just `CopySource`

Reply:
Oh you were thinking of reusing this class for other things. I see. Would it be better to keep this class only for CopySource purpose or not?

Reply:
I was thinking of using this class to hold the static APIs that do the requests transformations.
In this case we have two API `copyObject` and `UploadPartCopy`.
Since this utils is created to hold the CustomRequestTransformers thart will be used in Customizations I was thinking to keep its scope at this level rather than CopySource which is just a Field which gets transformed.